src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CrlCache.cs

```diff
@@ -91,13 +91,35 @@ private static bool AddCachedCrl(string crlFileName, SafeX509StoreHandle store,
                         return false;
                     }
 
-                    // If crl.LastUpdate is in the past, downloading a new version isn't really going
-                    // to help, since we can't rewind the Internet. So this is just going to fail, but
-                    // at least it can fail without using the network.
-                    //
-                    // If crl.NextUpdate is in the past, try downloading a newer version.
-                    DateTime nextUpdate = OpenSslX509CertificateReader.ExtractValidityDateTime(
-                        Interop.Crypto.GetX509CrlNextUpdate(crl));
+                    IntPtr nextUpdatePtr = Interop.Crypto.GetX509CrlNextUpdate(crl);
+                    DateTime nextUpdate;
+
+                    // If there is no crl.NextUpdate, this indicates that the CA is not providing
+                    // any more updates to the CRL, or they made a mistake not providing a NextUpdate.
+                    // We'll cache it for a few days to cover the case it was a mistake.
+                    if (nextUpdatePtr == IntPtr.Zero)
+                    {
+                        try
+                        {
+                            nextUpdate = File.GetLastWriteTime(crlFile).AddDays(3);
+                        }
+                        catch
+                        {
+                            // We couldn't determine when the CRL was last written to,
+                            // so consider it expired.
+                            Debug.Fail("Failed to get the last write time of the CRL file");
+                            return false;
+                        }
+                    }
+                    else
+                    {
+                        // If crl.LastUpdate is in the past, downloading a new version isn't really going
+                        // to help, since we can't rewind the Internet. So this is just going to fail, but
+                        // at least it can fail without using the network.
```

Comment:
This comment shouldn't really be moved to within this block here; since it's mainly talking about why we're ignoring lastUpdate... which is before we start looking at nextUpdate :smile:

Either restored to where it was, or deleted because it's confusing, but not here.